Assets/Scripts/Scenes/InGame/Enemy/EnemyMove.cs

```diff
@@ -69,138 +91,114 @@ void FixedUpdate()
 
             _staminaChangeCount += Time.deltaTime;
             if (_staminaChangeCount > 1)
-            {//毎秒処理
-                _staminaChangeCount -= 1;
+                 {//毎秒処理
+                    _staminaChangeCount -= 1;
+
+                switch (_enemyStatus.EnemyState)
+                {
+                    case EnemyState.Patrolling:
+                    case EnemyState.Searching:
+                        //通常の場合
+                        if (_enemyStatus.StaminaBase > _enemyStatus.Stamina)
+                        { //スタミナが削れていたら
+                            _enemyStatus.StaminaChange(_enemyStatus.Stamina + 1);
+                        }
+                        else if (_enemyStatus.StaminaBase < _enemyStatus.Stamina)
+                        {
+                            _enemyStatus.SetStuminaOver(false);
+                        }
+                        break;
+                    case EnemyState.Chase:
+                    case EnemyState.Attack:
+                        //走る場合
+                        if (_enemyStatus.StaminaOver)
+                        { //スタミナが切れ切ったかどうか
+                            if (_enemyStatus.StaminaBase <= _enemyStatus.Stamina && !(_enemyStatus.StaminaBase == 0))
+                            { //回復した状態にあるかどうか
+                                _enemyStatus.SetStuminaOver(false);
+                            }
+                            else
+                            {//回復しきっていないなら回復する
+                                if (_enemyStatus.StaminaBase > _enemyStatus.Stamina)
+                                { //スタミナが削れていたら、これがあるのはスタミナが0のキャラがいた時にまともに動かすため
+                                    _enemyStatus.StaminaChange(_enemyStatus.Stamina + 1);
+                                }
+                            }
+                        }
+                        else
+                        { //まだスタミナが切れ切って無い場合
+                            if (0 >= _enemyStatus.Stamina)
+                            { //たった今切れ切ったかどうか
+                                _enemyStatus.SetStuminaOver(true);
+                                if (_enemyStatus.StaminaBase > _enemyStatus.Stamina)
+                                { //スタミナが削れていたら、これがあるのはスタミナが0のキャラがいた時にまともに動かすため
+                                    _enemyStatus.StaminaChange(_enemyStatus.Stamina + 1);
+                                }
+                            }
+                            else
+                            {
+                                if (0 < _enemyStatus.Stamina)
+                                { //スタミナを削れるなら、これがあるのはスタミナが0のキャラがいた時にまともに動かすため
+                                    _enemyStatus.StaminaChange(_enemyStatus.Stamina - 1);
+                                }

```

Comment:
/
```suggestion
                                //スタミナを削れるなら、これがあるのはスタミナが0のキャラがいた時にまともに動かすため
                                    _enemyStatus.StaminaChange(_enemyStatus.Stamina - 1);

```
上のif文で0 >= _enemyStatus.Staminaがあり、elseでここに入ってるのこのif (0 < _enemyStatus.Stamina)いらないと思う。